test/python/transpiler/test_swap_strategy_router.py

```diff
@@ -497,6 +502,36 @@ def test_single_qubit_circuit(self):
 
         self.assertEqual(circ, self.pm_.run(circ))
 
+    @data(
+        ({(0, 1): 0, (2, 3): 0, (1, 2): 1}, 9),  # better coloring for the swap strategy
+        ({(0, 1): 1, (2, 3): 1, (1, 2): 0}, 11),  # worse, i.e., less CX cancellation.
+    )
+    @unpack
+    def test_edge_coloring(self, edge_coloring, cx_count):
+        """Test that the edge coloring works."""
+        op = PauliSumOp.from_list([("IIZZ", 1), ("IZZI", 2), ("ZZII", 3), ("ZIZI", 4)])
+        swaps = (((1, 2),),)
+
+        cmap = CouplingMap([[0, 1], [1, 2], [2, 3]])
+        cmap.make_symmetric()
+
+        swap_strat = SwapStrategy(cmap, swaps)
+
+        circ = QuantumCircuit(4)
+        circ.append(PauliEvolutionGate(op, 1), range(4))
+
+        pm_ = PassManager(
+            [
+                FindCommutingPauliEvolutions(),
+                Commuting2qGateRouter(swap_strat, edge_coloring=edge_coloring),
+                Decompose(),  # double decompose gets to CX
+                Decompose(),
+                CXCancellation(),
+            ]
+        )
+
+        self.assertEqual(pm_.run(circ).count_ops()["cx"], cx_count)
```

Comment:
This isn't a super convincing test to me - by reading it, I can't verify that the priorities of the colours are being respected (though I can see they have an effect).  Given that there's only one valid output and no randomness (as far as I could see) for the logical inputs, it might not hurt to construct the _exact_ output circuits and test against them.

Reply:
Good point. I added the expected circuits here: [43c87cf](https://github.com/Qiskit/qiskit-terra/pull/8341/commits/43c87cf34196c62b1338b0f4e50512545dc551b9)